mailbox/lucene/src/main/java/org/apache/james/mailbox/lucene/search/LenientImapSearchAnalyzer.java

```diff
@@ -47,7 +44,9 @@ public LenientImapSearchAnalyzer() {
     }
     
     @Override
-    public TokenStream tokenStream(String arg0, Reader reader) {
-        return new ShingleFilter(new UpperCaseFilter(new WhitespaceTokenizer(Version.LUCENE_31, reader)), 2, maxTokenLength);
+    protected TokenStreamComponents createComponents(String fieldName) {

```

Comment:
This whole class can be removed. Better use the builder pattern of CustomAnalyzer. Subclassing analyzers like that is no longer recommended.